multinet/api/views/serializers.py

```diff
@@ -40,6 +48,31 @@ class Meta:
         read_only_fields = ['created']
 
 
+class PermissionsSerializer(serializers.Serializer):
+    public = serializers.BooleanField()
+    owner = UserSerializer()
+    maintainers = UserSerializer(many=True)
+    writers = UserSerializer(many=True)
+    readers = UserSerializer(many=True)
+
+
+class PermissionsReturnSerializer(serializers.ModelSerializer):
+    owner = UserDetailSerializer()
+    maintainers = UserDetailSerializer(many=True)
+    writers = UserDetailSerializer(many=True)
+    readers = UserDetailSerializer(many=True)
+
+    class Meta:
+        model = Workspace
+        fields = WorkspaceCreateSerializer.Meta.fields + [
+            'public',
+            'owner',
+            'maintainers',
+            'writers',
+            'readers',
+        ]
```

Comment:
The `PermissionsReturnSerializer` contains many more details than the `PermissionsSerializer`, particularly about user records, which would be useful to have on the client. In the request serializer we only need a username to create the appropriate permission, so I think it makes sense to have two.

As to why its a `ModelSerializer`, it makes using it easier, since we have properties for readers, writers, etc. on the `Workspace` model, we can just pass in an instance and get all the information we need. I "extended" `WorkspaceCreateSerializer` because it seemed like a good starting point property-wise, and then added the permissions properties as well. I can see a case for this being too much information for a permissions request, in which case we could return a different subset of workspace properties. I still think we want two serializers though.

Let me know what your thoughts are on this or what you had in mind as a replacement.

Reply:
I think this approach is good. The only suggestions I have are:

1. Remove `public` from the field specified here. That's covered in `WorkspaceSerializer`
2. I think we should rename `PermissionsSerializer` to `PermissionsCreateSerializer`, to make things more clear.